packages/query/src/block-processor.ts

```diff
@@ -146,6 +146,65 @@ export class BlockProcessor implements BlockProcessorInterface {
       keepAlive: true,
       abortSignal: this.abortController.signal,
     })) {
+      let trial: undefined | string;
+      let epoch = false;
+
+      trialDecrypt: {
+        if (compactBlock.epochRoot) {
+          epoch = true;
+          break trialDecrypt;
+        }
+
+        const forgettable = new Array<Uint8Array>();
+        for (const statePayload of compactBlock.statePayloads) {
+          //console.log('state payload', statePayload);
+          switch (statePayload.statePayload.case) {
+            case 'note': {
+              const {
+                noteCommitment: { inner: commitmentVec } = { inner: undefined },
+                ephemeralKey: ephemeralKeyVec,
+                encryptedNote: { inner: encryptedNoteVec } = { inner: undefined },
+              } = statePayload.statePayload.value.note!;
+              //console.log({ commitmentVec, encryptedNoteVec, ephemeralKeyVec });
+              if (this.viewServer.canDecrypt(commitmentVec!, encryptedNoteVec!, ephemeralKeyVec)) {
+                trial = 'note';
+                break trialDecrypt;
+              } else forgettable.push(commitmentVec!);
+              break;
+            }
+            case 'swap': {
+              const {
+                commitment: { inner: commitmentVec } = { inner: undefined },
+                encryptedSwap: encryptedSwapVec,
+              } = statePayload.statePayload.value.swap!;
+              //console.log({ commitmentVec, encryptedSwapVec });
+              if (this.viewServer.canDecrypt(commitmentVec!, encryptedSwapVec)) {
+                trial = 'swap';
+                break trialDecrypt;
+              } else forgettable.push(commitmentVec!);
+              break;
+            }
+            case 'rolledUp': {
+              const commitment = statePayload.statePayload.value.commitment!;
+              const advice = await this.indexedDb.readAdvice(commitment);
+              if (advice) {
+                console.warn('readAdvice', advice);
+                trial = 'advice';
+                break trialDecrypt;
+              } else forgettable.push(commitment.inner);
+              break;
+            }
+            default: {
+              throw new Error('unknown state payload');
+            }
+          }
+        }
+
+        // didn't find anything. time to forget all of that
+        forgettable.map(commitment => this.viewServer.forgetCommitment(commitment));
+        this.viewServer.dontScanBlock(compactBlock.height);
```

Comment:
i'm only batching this because if we had found something, we would pass the compact block to wasm, which would progressively forget as it scans. so i have to avoid mutating the sct until i know the block is totally uninteresting